src/Swarm/Game/State.hs

```diff
@@ -823,8 +834,14 @@ buildWorld em (WorldDescription {..}) = (robots, first fromEnum . wf)
 initGameStateForScenario :: String -> Maybe Seed -> Maybe String -> ExceptT Text IO GameState
 initGameStateForScenario sceneName userSeed toRun = do
   g <- initGameState
-  scene <- loadScenario sceneName (g ^. entityMap)
-  liftIO $ scenarioToGameState scene userSeed toRun g
+  (scene, path) <- loadScenario sceneName (g ^. entityMap)
+  gs <- liftIO $ scenarioToGameState scene userSeed toRun g
+  normalPath <- liftIO $ normalizeScenarioPath (gs ^. scenarios) path
+  t <- liftIO getZonedTime
+  return $
+    gs
+      & currentScenarioPath ?~ normalPath
+      & scenarios . scenarioItemByPath normalPath . _SISingle . _2 . scenarioStatus .~ InProgress t 0 0
```

Comment:
I am unsure where, but we broke `--run`. For example:
```bash
swarm --run <(echo 'say "Hi!"')
```
works on the main but not this branch. :fearful:

Reply:
Found it! The option was not passed on in [`initAppState`.](https://github.com/swarm-game/swarm/pull/631/files#diff-e01da9c26d57a50bfdf719aaf9d0cf53e7f739146a8ec0db5bdea8fc304d3b07R861)